vj4/handler/problem.py

```diff
@@ -76,10 +76,22 @@ class ProblemDetailView(base.Handler):
   @base.sanitize
   async def post(self, *, pid: document.convert_doc_id, lang: str, code: str):
     pdoc = await problem.get(self.domain_id, pid)
-    rid = await record.add(self.domain_id, pdoc['doc_id'], self.user['_id'], lang, code)
+    rid = await record.add(self.domain_id, record.TYPE_SUBMISSION, pdoc['doc_id'], self.user['_id'], lang, code)
```

Comment:
If you want to add a `type` parameter, add it after `pid`. As `(domain_id, pid)` is correlated.